### PR TITLE
Maintenance events

### PR DESCRIPTION
**Pull request by Abyss777**
Here is implementation of "maintenance is needed" events.

Two configuration attributes:
- `device.maintenanceStart` zero if ommited 
- `device.maintenanceInterval` can't be zero
Both in meters

Example:
`device.maintenanceStart = 2000000`
`device.maintenanceInterval = 7500000`

Events will be generated once at the point then `totalDistance` overcome 2000 km, 9500 km, 17000 km, 24500 km and so on.

Also:
- Enabled `statusHandler` by default (or it was disabled intentionally? )
- Added mail templates for new event types

### CODE REVIEW

File: setup/default.xml [hunk 1]
```diff
@@ -24,6 +24,8 @@
     <entry key='event.geofenceHandler'>true</entry>
     <entry key='event.alertHandler'>true</entry>
     <entry key='event.ignitionHandler'>true</entry>
+    <entry key='event.statusHandler'>true</entry>
```
Comment (tananaev) [inline setup/default.xml]: Did we miss this?

File: src/org/traccar/events/MaintenanceEventHandler.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+    @Override
+    protected Collection<Event> analyzePosition(Position position) {
+        Device device = Context.getIdentityManager().getDeviceById(position.getDeviceId());
+        if (device == null || !Context.getIdentityManager().isLatestPosition(position)) {
+            return null;
+        }
+
+        double maintenanceInterval = Context.getDeviceManager()
+                .lookupServerDouble(device.getId(), ATTRIBUTE_MAINTENANCE_INTERVAL, 0);
```
Comment (tananaev) [inline src/org/traccar/events/MaintenanceEventHandler.java]: Not really related to this future, but maybe we should have just one set of "lookup" methods? And always pass parameter to it? Do you remember why we decided to have two sets?

File: src/org/traccar/events/MaintenanceEventHandler.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        double maintenanceStart = Context.getDeviceManager()
+                .lookupServerDouble(device.getId(), ATTRIBUTE_MAINTENANCE_START, 0);
+
+        Collection<Event> events = new ArrayList<>();
+        double oldTotalDistance = 0.0;
+        double newTotalDistance = 0.0;
+
+        Position lastPosition = Context.getIdentityManager().getLastPosition(position.getDeviceId());
+        if (lastPosition != null && lastPosition.getAttributes().containsKey(Position.KEY_TOTAL_DISTANCE)) {
+            oldTotalDistance = ((Number) lastPosition.getAttributes().get(Position.KEY_TOTAL_DISTANCE)).doubleValue();
```
Comment (tananaev) [inline src/org/traccar/events/MaintenanceEventHandler.java]: We use it quite a lot, so maybe we should add some helper methods to Position model to get ints, string, double etc?

File: src/org/traccar/model/Event.java [hunk 4]
```diff
@@ -56,6 +56,8 @@ public Event() {
     public static final String TYPE_IGNITION_ON = "ignitionOn";
     public static final String TYPE_IGNITION_OFF = "ignitionOff";
 
+    public static final String TYPE_MAINTENANCE_NEED = "maintenanceNeed";
```
Comment (tananaev) [inline src/org/traccar/model/Event.java]: I think we should just name it "maintenance" or alternatively "maintenance needed" (note -ed).

**Comment by tananaev** [comment]
Another tiny comment. We can probably rename "device.maintenanceStart" to "maintenance.start". I think it makes more sense.

File: src/org/traccar/events/MaintenanceEventHandler.java (hunk 2 above)
Comment (Abyss777) [inline src/org/traccar/events/MaintenanceEventHandler.java]: It was here https://github.com/tananaev/traccar/pull/2348#discussion_r79599245
and here https://github.com/tananaev/traccar/pull/2348#discussion_r79766522

File: src/org/traccar/events/MaintenanceEventHandler.java (hunk 3 above)
Comment (Abyss777) [inline src/org/traccar/events/MaintenanceEventHandler.java]: Yes, it must be useful. But It is better to make new pull request, and replace it everywhere.

Comment (tananaev) [inline src/org/traccar/events/MaintenanceEventHandler.java]: Sure.

File: src/org/traccar/events/MaintenanceEventHandler.java (hunk 2 above)
Comment (tananaev) [inline src/org/traccar/events/MaintenanceEventHandler.java]: Can we please change it to be one set of methods with boolean variable indicating whether to lookup config file or server model. Let's do it in another pull request, I'll merge this.

**Comment by tananaev** [comment]
Merged, thanks.

**Pull request merged**